platforms/android/src/jp/wizcorp/phonegap/plugin/WizAssets/HttpToFile.java

```diff
@@ -0,0 +1,133 @@
+package jp.wizcorp.phonegap.plugin.WizAssets;
+import java.io.*;
+import java.net.Authenticator;
+import java.net.HttpURLConnection;
+import java.net.PasswordAuthentication;
+import java.net.URL;
+import jp.wizcorp.phonegap.plugin.WizAssets.ILogger;
+
+public final class HttpToFile {
+    private static int _blockSize;
+    private static char[] _emptyCharArray = new char[0];
+    private static final String TAG = "WizAssetsPlugin";
+    private static ILogger _logger;
+
+    public static void setLogger(ILogger logger) {
+        _logger = logger;
+    }
+
+    public static void setBlockSize(int blockSize) {
+        _blockSize = blockSize;
+    }
+
+    public static boolean downloadFile(URL url, File file) throws IOException {
+        _logger.logDebug(TAG, "[Downloading to] " + file.getAbsolutePath());
+        BufferedInputStream inputStream = null;
+        HttpURLConnection urlConnection = null;
+        Boolean successfulWrite = false;
+        try {
+            if (!createPath(file)) {
+                _logger.logError(TAG, "file path error");
+            } else {
+                authenticate(url);
+                urlConnection = (HttpURLConnection) url.openConnection();
+                urlConnection = handleRedirect(urlConnection);
+                inputStream = new BufferedInputStream(urlConnection.getInputStream());
+                successfulWrite = writeFile(inputStream, file);
+            }
+        } catch (Exception e) {
+            printError(e);
+        } finally {
+            if (inputStream != null)
+                inputStream.close();
+            if (urlConnection != null) {
+                urlConnection.disconnect();
+            }
+        }
+        return successfulWrite;
+    }
+
+    private static HttpURLConnection handleRedirect(HttpURLConnection urlConnection) throws IOException {
+        int status = urlConnection.getResponseCode();
+        if (status != HttpURLConnection.HTTP_OK) {
+            if (status == HttpURLConnection.HTTP_MOVED_TEMP
+                    || status == HttpURLConnection.HTTP_MOVED_PERM
+                    || status == HttpURLConnection.HTTP_SEE_OTHER) {
+                urlConnection.disconnect();
+                String newUrl = urlConnection.getHeaderField("Location");
+                urlConnection = (HttpURLConnection) new URL(newUrl).openConnection();
+                _logger.logDebug(TAG, "Redirect to URL : " + newUrl);
+            }
+        }
+        _logger.logDebug(TAG, "Response Code ... " + status);
+        return urlConnection;
+    }
+
+    private static Boolean createPath(File file) {
+        File dir = file.getParentFile();
+        if (dir == null || !(dir.mkdirs() || dir.isDirectory())) {
+            _logger.logError(TAG, "Error: subdirectory could not be created");
+            return false;
+        }
+        return true;
+    }
+
+    private static Boolean writeFile(BufferedInputStream inputStream, File file) {
```

Comment:
Not a blocker: It would have been nice to have a method which takes both an inputStream and an outputStream.